src/retroshare/rsposted.h

```diff
@@ -213,16 +286,54 @@ class RsPosted : public RsGxsIfaceHelper, public RsGxsCommentService
 //virtual bool createNewComment(uint32_t &token, RsGxsComment &comment) = 0;
 //virtual bool createNewVote(uint32_t &token, RsGxsVote &vote) = 0;
 
+	/**
+	 * @brief toggle message read status
+	 * @deprecated
+	 * @param[out] token GXS token queue token
+	 * @param[in] msgId
+	 * @param[in] read
+	 */
     RS_DEPRECATED_FOR(setPostReadStatus)
     virtual void setMessageReadStatus(uint32_t& token, const RsGxsGrpMsgIdPair& msgId, bool read) = 0;

```

Comment:
This method is deprecated, i would not expose it. 
If the current WebUI requires this function, the WebUI should be fixed instead.

Reply:
i cloned only from the gxschannels class which has the same feature
https://github.com/RetroShare/libretroshare/blob/35360504c0ff85703a7e1b68d3261beda9612934/src/retroshare/rsgxschannels.h#L637

Reply:
But is this function really required for the webui?
Maybe we should fix gxschannels then :smiley:

Reply:
not sure only added :D

Reply:
remove it or keep this?

Reply:
remove it please, we can add it later or fix the webui